pandas/core/arrays/mask/_base.py

```diff
@@ -0,0 +1,126 @@
+"""A boolean mask interace.
+
+This module provides an interface to a numpy / pyarrow boolean mask.
+This is limited as not all of the implementations can hold NA, so
+for consistency this is an internal.
+"""
+
+import copy
+import numpy as np
+from pandas import compat
+from pandas.core.missing import isna
+from pandas.core.arrays.base import ExtensionArray
+from pandas.api.extensions import ExtensionDtype
+
+
+class BoolDtype(ExtensionDtype):
+
+    type = np.bool_
+    kind = 'b'
+    name = 'bool'
+
+    @classmethod
+    def construct_from_string(cls, string):
+        if string == cls.name:
+            return cls()
+        else:
+            raise TypeError("Cannot construct a '{}' from "
+                            "'{}'".format(cls, string))
+
+    def _is_boolean(self):
+        return True
+
+    def __hash__(self):
+        return hash(str(self))
+
+    def __eq__(self, other):
+        # compare == to np.dtype('bool')
+        if isinstance(other, compat.string_types):
+            return other == self.name
+        elif other is self:
```

Comment:
Maybe `isinstance(other, type(self))`? Then we get subclasses, and can maybe remove the `hash(self) == hash(other)`?